localstory.d.ts

```diff
@@ -4,20 +4,21 @@ declare module "localstory" {
     vacuum?: boolean;
   };
 
-  type LSStore = {
-    get: (key: string) => any;
-    set: (key: string, value: any, options?: LSOptions) => void;
-    unset: (key: string) => void;
-    has: (key: string) => boolean;
+  type LSStore<T> = {
+    get: (key: keyof T) => T[keyof T];
+    set: (key: keyof T, value: unknown, options?: LSOptions) => void;
+    unset: (key: keyof T) => void;
+    has: (key: keyof T) => boolean;
     clear: () => void;
     vacuum: () => void;
-    keys: () => [string];
-    values: () => [any];
+    keys: () => (keyof T)[];
+    values: () => (T[keyof T])[];
   };
 
-  export default function(
-    store: any,
-    ns?: string,
-    options?: LSOptions
-  ): LSStore;
+  type LSCreateStore<T extends object = { [key: string]: unknown }> =
+    (store: T, ns?: string, options?: LSOptions) => LSStore<T>;
```

Comment:
Usually you'll pass `window.localStorage` or `sessionStorage` as the store. How are you using this to set the type - shouldn't it take a generic so you can do:

```js
localstory<MyStoreShape>(window.localStorage, 'myNamespace')
```

Reply:
Yep, exactly this. `createStore: LSCreateStore` is the `localstory` function in your code snippet. It takes the store shape as a generic which should extend an object. 

Also after looking at this once again - probably it should be also defaulting to this kind object with unknown values when used without passing the generic type manually.

Reply:
@ricardobeat Updated it to expose a generic type to the users, missed it before as you've mentioned + improved method typings:

![image](https://user-images.githubusercontent.com/887379/94289458-089e8480-ff59-11ea-9406-ac8164d22ce3.png)